### ISSUE TITLE
GRSciColl - Endpoint for choosing/creating master source

### ISSUE THREAD

**Issue by ManonGros** [gbif/registry#399]
Relates to this issue: https://github.com/gbif/registry-console/issues/429

Right now a master source (= source with which a GRSciColl entry is synchronised) is handled with macthineTags. But most users don't have the permissions to create/delete matchineTags.
To allow them to choose a master sources (without them getting the machineTag namespace permissions), we need an endpoint. The idea is that:

- if a person has the permission to edit the core entity information, they should also be able to choose a master source
- in the case of IH, a person shouldn't be allowed to disconnect an entity from IH if this is the only connected entity. For example, you cannot disconnect a collection from IH otherwise a new collection will be created.

**Comment by marcos-lg** [comment]
We finally stopped using machine tags for the machine source and we created a new entity called `MasterSourceMetadata`. The endpoints are:

POST /institution/{key}/masterSourceMetadata

BODY:
{
  "source": "DATASET | ORGANIZATION | IH_IRN",
  "sourceId": "123"
}


GET /institution/{key}/masterSourceMetadata

DELETE /institution/{key}/masterSourceMetadata

The same for collections.

When removing an IH master source metadata, we check if it's the only entity connected to that IH IRN. If so, only admins can delete it.

They are deployed in DEV and UAT.
